scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/synchronization/SynchronizationInternal.java

```diff
@@ -0,0 +1,145 @@
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.synchronization;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.objectweb.proactive.annotation.PublicAPI;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 27/03/2018
+ */
+public interface SynchronizationInternal extends Serializable {
+
+    // Channel Operations
+
+    boolean createChannel(String originator, TaskId taskid, String name, boolean isPersistent) throws IOException;
+
+    boolean createChannelIfAbsent(String originator, TaskId taskid, String name, boolean isPersistent)
+            throws IOException;
+
+    boolean deleteChannel(String originator, TaskId taskid, String name) throws IOException;
+
+    // Standard map operations
+
+    int size(String originator, TaskId taskid, String channel) throws InvalidChannelException;
+
+    boolean isEmpty(String originator, TaskId taskid, String channel) throws InvalidChannelException;
+
+    boolean containsKey(String originator, TaskId taskid, String channel, String key) throws InvalidChannelException;
+
+    boolean containsValue(String originator, TaskId taskid, String channel, Serializable value)
+            throws InvalidChannelException;
+
+    <T extends Serializable> T get(String originator, TaskId taskid, String channel, String key)
+            throws InvalidChannelException;
+
+    <T extends Serializable> T put(String originator, TaskId taskid, String channel, String key, T value)
+            throws InvalidChannelException, IOException;
+
+    <T extends Serializable> T remove(String originator, TaskId taskid, String channel, String key)
+            throws InvalidChannelException, IOException;
+
+    <T extends Serializable> void putAll(String originator, TaskId taskid, String channel,
+            Map<? extends String, ? extends T> m) throws InvalidChannelException, IOException;
+
+    void clear(String originator, TaskId taskid, String channel) throws InvalidChannelException, IOException;
+
+    Set<String> keySet(String originator, TaskId taskid, String channel) throws InvalidChannelException;
+
+    <T extends Serializable> Collection<T> values(String originator, TaskId taskid, String channel)
+            throws InvalidChannelException;
+
+    <T extends Serializable> Set<Map.Entry<String, T>> entrySet(String originator, TaskId taskid, String channel)
+            throws InvalidChannelException;
+
+    <T extends Serializable> T getOrDefault(String originator, TaskId taskid, String channel, String key,
+            T defaultValue) throws InvalidChannelException;
+
+    void forEach(String originator, TaskId taskid, String channel, String action)
+            throws InvalidChannelException, CompilationException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)